core/src/fields.rs

```diff
@@ -469,50 +469,150 @@ impl MongoFields {
+                }
+            }
+            if self.dbs.is_empty() {
+                return false;
+            }
+            let db_name = self.dbs.remove(0);
+            let db = mongo_connection.client.database(&db_name);
+            self.current_db_name = db_name;
+            self.collections_for_db = Some(
+                db.list_collections(self.collection_name_filter.clone(), None)
+                    .unwrap(),
+            );
         }
     }
 }
 
 impl MongoStatement for MongoFields {
     // Move the cursor to the next document and update the current row.
     // Return true if moving was successful, false otherwise.
-    fn next(&mut self, _mongo_connection: Option<&MongoConnection>) -> Result<bool> {
-        unimplemented!()
+    fn next(&mut self, mongo_connection: Option<&MongoConnection>) -> Result<bool> {
+        self.current_field_for_collection += 1;
+        Ok(
+            (self.current_field_for_collection as usize) < self.current_col_metadata.len()
+                || self.get_next_metadata(mongo_connection.unwrap()),
+        )
     }
 
     // Get the BSON value for the cell at the given colIndex on the current row.
     // Fails if the first row as not been retrieved (next must be called at least once before getValue).
-    fn get_value(&self, _col_index: u16) -> Result<Option<Bson>> {
-        unimplemented!()
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
+        // use self.current_col_metadata[current_field_for_collection]
+        // 1 -> TABLE_CAT
+        // 2 -> TABLE_SCHEM  (NULL)
+        // 3 -> TABLE_NAME
+        // 4 -> COLUMN_NAME
+        // 5 -> DATA_TYPE
+        // 6 -> TYPE_NAME
+        // 7 -> COLUMN_SIZE
+        // 8 -> BUFFER_LENGTH
+        // 9 -> DECIMAL_DIGITS
+        // 10 -> NUM_PREC_RADIX
+        // 11 -> NULLABLE
+        // 12 -> REMARKS
+        // 13 -> COLUMN_DEF
+        // 14 -> SQL_DATA_TYPE
+        // 15 -> SQL_DATETIME_SUB
+        // 16 -> CHAR_OCTET_LENGTH
+        // 17 -> ORDINAL_POSITION
+        // 18 -> IS_NULLABLE "YES" or "NO"
+        let get_meta_data = || {
+            self.current_col_metadata
+                .get(self.current_field_for_collection)
+                .unwrap()
+        };
+        Ok(Some(match col_index {
+            1 => Bson::String(self.current_db_name.clone()),
+            2 => Bson::Null,
+            3 => Bson::String(get_meta_data().table_name.clone()),
+            4 => Bson::String(get_meta_data().col_name.clone()),
+            5 => Bson::Int32(get_meta_data().sql_type.0 as i32),
+            6 => Bson::String(get_meta_data().type_name.clone()),
+            7 => Bson::Int32(get_meta_data().length.unwrap_or(0) as i32),
+            8 => Bson::Int32(get_meta_data().length.unwrap_or(0) as i32),
+            9 => Bson::Int32(get_meta_data().precision.unwrap_or(0) as i32),
+            10 => match get_meta_data().sql_type {
+                SqlDataType::INTEGER | SqlDataType::DOUBLE => Bson::Int32(2),
```

Comment:
it's a binary number

Reply:
I'll leave everything as 10. It seems INTEGER could be 2 or 10 (it doesn't matter since there is no exponent, 2^0 = 10^0). But technically double uses 2 as a radix while decimal uses 10. I'm not sure if that's what they want?

Reply:
ah, it might be what it is about or what it was supposed to be but ended up being something else... 
The spec does not say and does not seem to care.

This is the excerpt from [the spec](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function?) 
> For example, a DECIMAL(12,5) column would return a NUM_PREC_RADIX of 10, a COLUMN_SIZE of 12, and a DECIMAL_DIGITS of 5; a FLOAT column could return a NUM_PREC_RADIX of 10, a COLUMN_SIZE of 15, and a DECIMAL_DIGITS of NULL.

If it is 2, the values in COLUMN_SIZE and DECIMAL_DIGITS give the number of bits allowed in the column. For example, a FLOAT column could return a RADIX of 2, a COLUMN_SIZE of 53, and a DECIMAL_DIGITS of NULL.